### PR TITLE
Fix: topic with one partition cannot be updated

### PR DESCRIPTION
**Pull request by jerrypeng**
### Motivation

Previously, we add the the following change to allow partitioned topics to be created with one partition:

https://github.com/apache/pulsar/pull/4764

However, updating a partitioned topic with one partition to more than one partition does not work and will return the following error:

: javax.ws.rs.ClientErrorException: HTTP 409 Conflict
Topic is not partitioned topic

Reason: Topic is not partitioned topic

### CODE REVIEW

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java [hunk 1]
```diff
@@ -2162,11 +2162,6 @@ private PersistentReplicator getReplicatorReference(String replName, PersistentT
         String path = path(PARTITIONED_TOPIC_PATH_ZNODE, topicName.getPersistenceNamingEncoding());
         CompletableFuture<Void> result = new CompletableFuture<>();
         pulsar().getBrokerService().fetchPartitionedTopicMetadataAsync(topicName).thenAccept(partitionMetadata -> {
-            if (partitionMetadata.partitions <= 1) {
```
Comment (merlimat) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: Shouldn't this be changed into `< 1` ?

Comment (jerrypeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: partitionMetadata is the current metadata for the topic.  If we check "< 1" that would be saying for any topic that has greater than one partition, it cannot be updated and the exception will be returned.

Comment (merlimat) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: No, it means to throw exception if `partitionMetadata.partitions == 0` (which is the case when topic is not partitioned), though not throwing it when `partitionMetadata.partitions == 1` (as we're incorrectly doing now) or when `partitionMetadata.partitions > 1` (which is already correctly handled).

Comment (jerrypeng) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/admin/impl/PersistentTopicsBase.java]: will fix

**Pull request merged**